libvast/src/system/index.cpp

```diff
@@ -1341,20 +1412,17 @@ index(index_actor::stateful_pointer<index_state> self,
                           relational_operator::ni, data{""}};
       auto query
         = query::make_extract(sink, query::extract::drop_ids, match_everything);
-      auto query_id = self->state.create_query_id();
-      auto lookup
-        = query_state{query_id, query, old_partition_ids, std::move(*worker)};
-      auto actors
-        = self->state.collect_query_actors(lookup, old_partition_ids.size());
-      if (actors.empty()) {
-        self->send(self, atom::worker_v, lookup.worker);
-        return caf::make_error(ec::invalid_argument,
-                               fmt::format("all partition ids invalid: {}",
-                                           old_partition_ids));
-      }
-      self->send(lookup.worker, atom::supervise_v, query_id, lookup.query,
-                 std::move(actors),
-                 caf::actor_cast<receiver_actor<atom::done>>(sink));
+      query.id = self->state.pending_queries.create_query_id();
+      query.priority = 100;
+      auto input_size = detail::narrow_cast<uint32_t>(old_partition_ids.size());
+      auto err = self->state.pending_queries.insert(
+        query_state{.query = query,
+                    .client = caf::actor_cast<receiver_actor<atom::done>>(sink),
+                    .candidate_partitions = input_size,
+                    .requested_partitions = input_size},
+        std::vector{old_partition_ids});
+      VAST_ASSERT(err == caf::none);
```

Comment:
Why can't this fail?

Reply:
We use the `create_query_id()` function that already deals with collisions. That is currently the only failure mode.